frontend/javascripts/oxalis/api/cross_origin_api.js

```diff
@@ -0,0 +1,34 @@
+// @flow
+import { useEffect } from "react";
+import _ from "lodash";
+
+import api from "oxalis/api/internal_api";
+
+// This component allows cross origin communication, for example, between a host page
+// and an embedded webKnossos iframe.
+// Currently, this is only used to set a mapping, but the interface may be extended in the future
+
+const onMessage = event => {
+  // We could use this to restrict usage of this api to specific domains
+  // if (event.origin !== "https://contactome-viewer.org") {
+  //   return;
+  // }
+  if (!_.isObject(event.data)) return;
+  const { type, args } = event.data;
+  if (type == null || !_.isArray(args)) return;
+
+  if (type === "setMapping") {
+    api.data.setMapping(api.data.getVolumeTracingLayerName(), ...args);
+  }
+};
+
+const CrossOriginApi = () => {
+  useEffect(() => {
+    window.addEventListener("message", onMessage);
+    return () => window.removeEventListener("message", onMessage);
+  });
```

Comment:
Shouldn't we pass an empty dependency array here as another parameter? Otherwise, the handler is re-attached unnecessarily everytime the parent component updates?

Reply:
:see_no_evil: Absolutely!